mdflibrary/CMakeLists.txt

```diff
@@ -1,11 +1,23 @@
 # Copyright 2023 Simplxs
 # SPDX-License-Identifier: MIT
+project(
+  mdflibrary
+  VERSION 2.0
+  DESCRIPTION "Interface against MDF 3/4 files"
+  LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_DEBUG_POSTFIX d)
+
 message(STATUS  "Compiler Version " ${CMAKE_CXX_COMPILER_VERSION})
 message(STATUS  "MSVC " ${MSVC})
-if (MSVC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.36)
+if (MDF_BUILD_SHARED_LIB_NET)
     message(STATUS  "CLR compile")
     add_library(mdflibrary SHARED
-            src/MdfExport.cpp src/MdfExport.h
+            # C++ files
+            src/MdfExport.cpp
```

Comment:
According to the CMake manual, you should add library as a MODULE. If so, you dont't need the Export.cpp file. The -NET have all export function. Well worth a try at least

Reply:
Cannot. `src/MdfExport.cpp` contains functions that need to be exported. If change SHARED to MODULE, cmake would not export symbols to dll.
As the CMake manual said.
"For example, a Windows resource DLL or a managed C++/CLI DLL that ***exports no unmanaged symbols*** would need to be a MODULE library."

Reply:
I trid to change SHARED to MODULE, and cmake complain that `Target "mdflibrary" of type MODULE_LIBRARY may not be linked into another target. One may link only to INTERFACE, OBJECT, STATIC or SHARED libraries, or to executables with the ENABLE_EXPORTS property set.`